### PR TITLE
Branch update dg

### PR DESCRIPTION
**Pull request by NeoMinWei**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -185,20 +182,78 @@ switch (category) {
...
-![](images/AddCashflowSequence.png)
+![](images/cashflow/AddCashflowSequence.png)
+### Recurring Cashflow Feature
+Cashflow refers to an income or expense.
+
+This feature is called from the user through the `/r` argument in add income/expense command.
+
+If a cashflow is set to be recurring, the program would add another entry of the same cashflow to the Financial Planner after a set period of time.
+
+Below are the steps that shows the implementation of the recurring cashflow feature.
```
Comment (ryan1604) [inline docs/DeveloperGuide.md:205]: show*

File: src/main/java/seedu/financialplanner/storage/LoadData.java [hunk 2]
```diff
@@ -140,12 +140,16 @@ private static void identifyRecurredCashflows(LocalDate currentDate
...
-            tempCashflow.add(toAdd);
+            addToTempList(tempCashflowList, toAdd);
             cashflow = toAdd;
             dateOfAddition = dateOfAddition.plusDays(recur);
         }
     }
 
+    private static void addToTempList(ArrayList<Cashflow> tempCashflowList, Cashflow toAdd) {
+        tempCashflowList.add(toAdd);
+    }
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/storage/LoadData.java:151]: Not sure the point of this method

File: src/main/java/seedu/financialplanner/storage/LoadData.java [hunk 3]
```diff
@@ -104,31 +104,31 @@ private static void deleteFutureCashflows(LocalDate currentDate) {
...
         }
-        for (Cashflow cashflow : tempCashflow) {
+        for (Cashflow cashflow : tempCashflowList) {
             cashflowList.load(cashflow);
             ui.printAddedCashflow(cashflow);
         }
     }
 
-    private static void addRecurringCashflowToTempList(LocalDate currentDate
+    private static void identifyRecurringCashflows(LocalDate currentDate
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/storage/LoadData.java:120]: Good job renaming to clearer names

File: src/main/java/seedu/financialplanner/cashflow/CashflowList.java [hunk 4]
```diff
@@ -29,19 +29,23 @@ public void addIncome(double value, IncomeType type, int recur, String descripti
...
+        addToList(toAdd);
         ui.printAddedCashflow(toAdd);
 
         int newListSize = list.size();
         assert newListSize == existingListSize + 1;
     }
 
+    private void addToList(Cashflow toAdd) {
+        list.add(toAdd);
+    }
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/cashflow/CashflowList.java:41]: Why add this method?

File: docs/diagrams/cashflow/CashflowClassDiagram.puml
Comment (ryan1604) [inline docs/diagrams/cashflow/CashflowClassDiagram.puml:1]: Can consider omitting less important details in both class diagrams

**Review by ryan1604 — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -185,20 +182,78 @@ switch (category) {
...
+Example:
+```
+public abstract class Cashflow {
+    protected int recur;
+    protected LocalDate date;
+    protected boolean hasRecurred;
+}
+```
+#### Step 2
+When the Financial Planner is started again in the future, the date of startup would be collected.
```
Comment (wwweert123) [inline docs/DeveloperGuide.md]: obtained from the system (?)

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -185,20 +182,78 @@ switch (category) {
...
+Once a cashflow matches the above criteria, the date of its next addition to the Financial Planner, `dateOfAddition`, would be determined.
+
+`dateOfAddition` would be compared to the current date, and if the current date is after or equal to `dateOfAddition`, an identical cashflow would be instantiated.
+
+This identical cashflow would then have its `date` variable set to `dateOfAddition`, then this cashflow would be added to a temporary list, `tempCashflowList`.
+
+The original cashflow would then have its `hasRecurred` variable be set to `true`.
+
+#### Step 4
+Each cashflow in `tempCashflowList` goes through **Step 3** again, so that multiple cashflows can be added if it has recurred more than once.
```
Comment (wwweert123) [inline docs/DeveloperGuide.md:244]: This can be phrased better i think

File: docs/diagrams/cashflow/CashflowClassDiagram.puml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    DINING
+    ENTERTAINMENT
+    SHOPPING
+    TRAVEL
+    INSURANCE
+    NECESSITIES
+    OTHERS
+}
+
+enum "<<enumeration>>\nIncomeType" as IncomeType #MintCream{
```
Comment (wwweert123) [inline docs/diagrams/cashflow/CashflowClassDiagram.puml:53]: Is there suppose to be a dependency on the enum?

File: src/main/java/seedu/financialplanner/storage/LoadData.java [hunk 8]
```diff
@@ -80,21 +80,21 @@ public static void load(String filePath, LocalDate date) throws FinancialPlanner
     }
 
     private static void deleteFutureCashflows(LocalDate currentDate) {
-        ArrayList<Integer> tempCashflow = new ArrayList<>();
+        ArrayList<Integer> tempCashflowList = new ArrayList<>();
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/storage/LoadData.java:83]: array list naming convention probably tempCashflows

**Review by wwweert123 — APPROVED** [review]
LGTM

File: src/main/java/seedu/financialplanner/cashflow/CashflowList.java (hunk 4 above)
Comment (NeoMinWei) [reply src/main/java/seedu/financialplanner/cashflow/CashflowList.java:41]: key to success

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (NeoMinWei) [reply docs/DeveloperGuide.md:244]: how

File: src/main/java/seedu/financialplanner/storage/LoadData.java (hunk 8 above)
Comment (NeoMinWei) [reply src/main/java/seedu/financialplanner/storage/LoadData.java:83]: from where

Comment (ryan1604) [reply src/main/java/seedu/financialplanner/storage/LoadData.java:83]: Both same

Comment (wwweert123) [reply src/main/java/seedu/financialplanner/storage/LoadData.java:83]: tempCashflowList -> tempCashflows

**Pull request merged**